src/main/java/seedu/mtracker/MTracker.java

```diff
@@ -15,7 +18,16 @@ public MTracker() {
 
     public void run() {
         TextUi.greetAtStartUp();
-
+        String userInput = InputParser.getUserInput();
+        String[] components = InputParser.getCommandComponents(userInput);
+        try {
+            Command c = InputParser.filterByCommandType(components);
```

Comment:
```suggestion
            Command command = InputParser.filterByCommandType(components);
            command.setData(instrumentManager)
```

Rename c to command to make it more explicit. You also need to setData so that the command object has access to the instance of instrumentManager so that all executions is done on that instrumentManager.

Do test out locally that you add in the instruments when you run the code

Reply:
Ok, setData will be coded in for issue # 29